identity-iota/src/did/doc/iota_document.rs

```diff
@@ -67,38 +67,30 @@ impl TryMethod for IotaDocument {
 }
 
 impl IotaDocument {
-  /// Creates a new DID Document from the given KeyPair.
-  ///
-  /// The DID Document will be pre-populated with a single authentication
-  /// method based on the provided [KeyPair].
-  ///
-  /// The authentication method will have the DID URL fragment `#authentication`
-  /// and can be easily retrieved with [Document::authentication].
-  pub fn from_keypair(keypair: &KeyPair) -> Result<Self> {
-    let method: IotaVerificationMethod = IotaVerificationMethod::from_keypair(keypair, "authentication")?;
-
-    // SAFETY: We don't create invalid Methods.  Method::from_keypair() uses the MethodBuilder
-    // internally which verifies correctness on construction.
-    Ok(unsafe { Self::from_authentication_unchecked(method) })
-  }
+  const DEFAULT_METHOD_FRAGMENT: &'static str = "authentication";
```

Comment:
I'm wondering a bit about the discrepancy between what the default account-generated document and the default `IotaDocument` look like. The low-level API uses this `"authentication"` fragment, while the account uses `"_sign-0"` as the default fragment.

https://github.com/iotaledger/identity.rs/blob/31e073e3694e59ac29333568ac8e8049e8eac784/identity-account/src/types/key_location.rs#L23

I don't see why they should not be consistent, and if we want to make them so, I would rather change the account to use something a little easier to type. But with the changes in #421 calling the default verification method `"authentication"` isn't quite right either, because it's a general verification method at that point. Since its main purpose is signing, I would suggest to use `"sign"` as the default in the low-level API, and `"sign-0"` in the account. Any thoughts on this? (This might be out of scope for this PR.)

Reply:
There's also the default verification method fragment `"key"`:

https://github.com/iotaledger/identity.rs/blob/73f8aa754bc22e7e7a7107f9a622c18c21d1c403/identity-iota/src/did/doc/iota_verification_method.rs#L38

I agree that the `Account` should default to the same name as the low-level API. I particularly dislike how the fragment is used to determine whether or not a verification method is used for authentication in [`KeyLocation`](https://github.com/iotaledger/identity.rs/blob/31e073e3694e59ac29333568ac8e8049e8eac784/identity-account/src/types/key_location.rs#L57):

```
  /// Returns true if the key location points to an authentication method.
  pub fn is_authentication(&self) -> bool {
    Self::is_authentication_fragment(&self.fragment)
  }

  /// Returns true if the fragment points to an authentication method.
  pub fn is_authentication_fragment(fragment: &Fragment) -> bool {
    fragment.name().starts_with(Self::AUTH)         // <============= "_sign-"
  }
```

This could lead to confusion if someone happens to name their key `"_sign-xyx"`? Probably out-of-scope for this PR.

While the default key is used for signing, it's a bit more special than a regular verification method since it can sign document updates too. The two choices are either: choose a special name like `"auth"`/`"default"`/`"bikeshedding"`, or treat it as a regular verification method, like `"key-0"` or `"sign-0"`. Probably worth getting some more opinions on this.

Reply:
The IOTA DID specification recommends the `"_sign-x"` name for verification method used to update the document. Just a reminder to apply whatever the final decision is to the documentation too.

> It is RECOMMENDED to name this public key #_sign-x, where x is the index of the signing key, which is incremented every time the signing key is updated, starting at index 1. 


https://github.com/iotaledger/identity.rs/blob/73f8aa754bc22e7e7a7107f9a622c18c21d1c403/documentation/docs/specs/iota_did_method_spec.md#L102

Reply:
Looks like we're going with `"sign-x"`.

Reply:
Unfortunately verification method names are tightly coupled with the logic in the Account, so it's not as simple as choosing a different name and using any `authentication` verification method. It needs its own issue/PR.